### DISCUSSION TITLE
Why does GitHub recommend closing issues before archiving a repository?

### DISCUSSION THREAD

**Discussion by pdpinch** [community/community#22554] [New to GitHub]
<p>The <a href="https://help.github.com/articles/about-archiving-repositories/" rel="nofollow noopener">GitHub documentation on archiving repositories</a> recommends closing all open issues before archiving a repository, but doesn’t explain why.</p>
<p>I understand that issues cannot be updated once a repository is archived, but it seems like a useful thing to know what unresolved issues there are be in an archived repository.</p>
<p>Since I <em>can</em> archve a repository without closing the issues, what are the downsides?</p>

**Comment by Wabri** [comment]
<p>There is no real reason, I think it is for completeness. If you have archive a repository you have probably solved the whole set of issues, otherwise you haven’t solved the purpose of the repository. It is more a philosophical concept than a rule.</p>

**Comment by CleanCut** [comment, marked as answer]
<p>Thanks for asking!  I asked around in engineering and found the person who came up with that recommendation.</p>
<p>According to him, there was a point in time when we were considering whether to auto-close all issues and pull requests when a repository was archived.  We felt (and still feel) that not closing issues and pull requests before archiving a repository has drawbacks, foremost among them:</p>
<ul>
<li>
<p>People often assume that open issues and pull requests will eventually be acted upon</p>
</li>
<li>
<p>The person who opened the issue will forever have it listed as open, and will be unable to close it themselves.</p>
</li>
</ul>
<p>In the end, it was decided to make it a recommendation rather than an automatic behavior.  I submitted a suggestion to the documentation team to add this reasoning to that help document for clarity.</p>

**Reply by tukusejssirs** [reply]
IMHO, the issues and PRs should stay open after archiving a repo, but all issues and PRs (regardless if they are open/closed/merged/rejected) should be **locked**.

In my opinion, if the open issues are closed before repo archival, it would seem to future bygoers that those issues were actually fixed before the repo was archived.

**Reply by CleanCut** [reply]
@tukusejssirs Use the "Close as not planned" button to indicate that the issue/PR was not fixed before it was closed.

<img width="429" alt="image" src="https://user-images.githubusercontent.com/5838512/210091875-27fe8a9d-1c63-4ca7-835b-cc5f6038b93e.png">

**Reply by allan2** [reply]
@CleanCut Thanks for the answer!

Would it be possible to request a feature for an option to auto-close all issues and PRs when archiving?

I had to close over 250 issues and PRs when archiving [BloomRPC](https://github.com/bloomrpc/bloomrpc). I ended up making a [program to do it through the API](https://github.com/allan2/github-issue-closer), but I feel like this is a lot of effort and many people will end up ignoring the recommendation to close because, lazy.

**Reply by CleanCut** [reply]
@allan2 Sure thing, you can make the feature request [here](https://github.com/orgs/community/discussions/new?category=pull-requests).

**Reply by spenserblack** [reply]
> IMHO, the issues and PRs should stay open after archiving a repo, but all issues and PRs (regardless if they are open/closed/merged/rejected) should be **locked**.

Agreed. I feel like keeping issues open makes it visible what is still pending, allowing a potential forker to pick the work back up (or the owner could start up again if they un-archive the project).
If all open issues are closed as "not planned" when the repo is archived, I feel like it could be a little ambiguous: were they not planned because the owner has no plans to manage the repo anymore, or because they just weren't "good" issues for the repo (duplicates, wontfix, etc.). In other words, if I forked a archived repo that closed all outstanding issues as "not planned", it wouldn't be easy to find the issues I could reopen on vs. the issues that should *stay* "not planned." I think the only hint would be if the closed date was around the same time the repo was archived.

I guess what I'm saying is that open issues on archived repos might be annoying for issue authors, assignees, and repo owners, but it adds a bit of helpful context for people that want to continue the work that the archived repo started.

**Reply by ghost** [reply]
scusate, io non sono completamente d'accordo. Ad  otto bre ho chiesto una mano e mi sono trovato in git-ignore senza una risposta. ho cors lasciato 5 pc di Lavoro. 
la notte scorsa oer una stupidaggine di Chrome che non andava, ni hanno aiutato, ma avendo problemi con la lingua ho chiesto gentilmente di nn essere abbandonato fino all'ultimo.. Ebbene. m'hanno lasciato con una stringa in mano che nn sapevo manco io dove mettere. ho scrittto una mail a chi mi ha aiutato ma nulla, si sono accorti solo allora che gli altri 5 pc erano stati formattati andando a perdere circa 8000 cartelle cliniche. 
ci sono fijito io di fronte al giudice.  O ggi con "westrem gentilezza" e senza alcun rabcore alla fine ho posizionto la tringa dove pensavo fose giusto. 
MAI LO AVESSI FATTO. hackerato il pc, prese foto personali e fatto achhe screenshoth che è illegale. sono rimasto 1 notte interae fino alle 19 di sera cercando di capire che fare. ùù
Sembravano delle iene contro di me. Li ho lasciati fare xkè non volevano neanche che gli aiutassi a ridargli quei due codici. stavolta però ho registrato tutto con dispositivi non associati. compresi gli insulti OMOFOBI . sono solo in attesa di un CEO. intanto l Polizia Postale è giò stata avvertita! sarò un newbby, ma di cerrto nn vorrei esssere come loro. AMEN

**Reply by ghost** [reply]
nah, vedi si stann anccora divertendo a guardarmi in cam, ad ascoltare, a cancellare i ost e spegnere il pc....

**Comment by wolph** [comment]
I just ran into this annoyance today, but from the other end :)
Once in a while I'll look at all of my open issues by using this page: https://github.com/issues

And unfortunately, by default, the archived issues show up there as well which is a tad annoying. So now I have to unarchive my repos so I can close the issues ;)

**Reply by spenserblack** [reply]
GitHub currently has 1 open status and 2 closed statuses (closed, not planned). I wonder if another open status, something like `open (archived)` makes sense 🤔

**Comment by wolph** [comment]
> GitHub currently has 1 open status and 2 closed statuses (closed, not planned). I wonder if another open status, something like `open (archived)` makes sense 🤔

I think it would make more sense to have this as an extra filter. Similar to how we can specify public/private visibility, archived should be an option as well

For the time being, here's how I solved it:
```zsh
# Loop through all archived repos
for repo in $(gh repo list --archived | cut -f 1); do
 # Unarchive the repo
 gh repo unarchive -y $repo
 # Sleep a bit because github takes some time to update the archived status
 sleep 5
 # Get all open issues/pull-requests and close them all
 gh issue list --state open -R $repo | cut -f 1 | xargs -L1 gh issue close -R $repo
 gh pr list --state open -R $repo | cut -f 1 | xargs -L1 gh pr close -R $repo
 # Re-archive the repo
 gh repo archive -y $repo
done
```
